ecl/eclcc/WORKUNITS.rst

```diff
@@ -0,0 +1,862 @@
+* Meta information the code generator has deduced about the activity output.  Examples include the 
+  record size, expected number of rows, sort order etc.  E.g., <att name="recordSize" value="120"/>
+* Hints, which are used for fine control of options for a particular activity (e.g,, the number of threads to use while sorting).
+* Record counts and stats once the job has executed.  (These are logically associated with the activities in the graph, but stored separately.)
+
+Graphs also contain edges which can take one of 3 forms:
+
+Edges within graphs
+    These are used to indicate how the activities are connected.  The source activity is used as the input to the target activity.
+    These edges have the following format::
+    
+        <edge id="<source-activity-id>_<output-count>" source="<source-activity-id>" target="<target-activity-id">
+    
+    There is only one edge in our example workunit: <edge id="2_0" source="2" target="3"/>.
+    
+Edges between graphs
+    These are used to indicate direct dependencies between activities.  For instance there will be an edge connecting the activity that
+    writes a spill file to the activity that reads it.  These edges have the following format::
+
+        <edge id="<source-activity-id>_<target-activity-id>" source="<source-subgraph-id>" target="<target-subgraph-id>"
+           <att name="_sourceActivity" value="<source-activity-id>"/>
+           <att name="_targetActivity" value="<target-activity-id>"/>
+        </edge>
+    
+    Roxie often optimizes spilled datasets and treats these edges as equivalent to the edges between activities.
+        
+Other dependencies.
+    These are similar to the edges between graphs, but they are used for values that are used within an activity.  For
+    instance one part of the graph may calculate the maximum value of a column, and another activity may filter out
+    all records that don't match that maximum value.  The format is the same as the edges between graphs except the edge contains
+    the following attribute::
+    
+        <att name="_dependsOn" value="1"/>
+
+Each activity in a graph also has a corresponding helper class instance in the generated code.  (The
+name of the class is "cAc" followed by the activity number, and the exported factory method is "fAc"
+followed by the activity number.)  Each helper class implements a specialised interface (derived from IHThorArg) - the
+particular interface is determined by the value of the "_kind" attribute for the activity.
+
+The contents of file rtl/include/eclhelper.hpp is key to understanding how the generated code relates to the activities.
+Each kind of activity requires a helper class that implements a specific interface.  The helpers allow the engine to
+tailor the generalised activity implementation to the the particular instance - e.g., for a filter activity
+whether a row should be included or excluded.  The appendix contains some further information about this file.
```

Comment:
You may need to be specific that you are referring to the appendix of this doc rather than something else.